### ISSUE TITLE
Update `check_connection` method

### ISSUE THREAD

**Issue by vfdev-5** [facebookresearch/visdom#452]
In two python example provided in the repository we can observe a similar pattern for the connection checking:
```
    startup_sec = 1
    while not viz.check_connection() and startup_sec > 0:
        time.sleep(0.1)
        startup_sec -= 0.1
    assert viz.check_connection(), 'No connection could be formed quickly'
```

I wonder whether this could not be integrated directly to `check_connection` method without asking the user to write such loop.

What do you think ?

Thanks

**Comment by vfdev-5** [comment]
I have also an issue with example demo.py when I run
```
python3 -m visdom.server &
python3 demo.py
```
I get the following output:
```
INFO:tornado.access:200 POST /env/main (127.0.0.1) 1.17ms
INFO:tornado.access:101 GET /vis_socket (127.0.0.1) 0.99ms
INFO:root:Opened visdom socket from ip: 127.0.0.1
unorderable types: float() > NoneType()
INFO:tornado.access:101 GET /vis_socket (127.0.0.1) 0.87ms
INFO:root:Opened visdom socket from ip: 127.0.0.1
unorderable types: float() > NoneType()
INFO:tornado.access:101 GET /vis_socket (127.0.0.1) 0.92ms
INFO:root:Opened visdom socket from ip: 127.0.0.1
unorderable types: float() > NoneType()
Visdom python client failed to establish socket to get messages from the server. This feature is optional and can be disabled by initializing Visdom with `use_incoming_socket=False`, which will prevent waiting for this request to timeout.
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.87ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.86ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.88ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.84ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.83ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.85ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.88ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.83ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.83ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.84ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.87ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.90ms
INFO:tornado.access:200 POST /win_exists (127.0.0.1) 0.84ms
The visdom experienced an exception while running: AssertionError('No connection could be formed quickly',)
```
However if comment connection checking:
```
# assert viz.check_connection(), 'No connection could be formed quickly'
```
demo works

**Comment by JackUrb** [comment]
@vfdev-5 It seems the latter issue is related to #450 - I'm not quite sure what the root cause of that is.

For your former suggestion, I'd be happy to accept a PR that moves that timeout logic behind a `timeout` argument to `check_connection()`

**Comment by vfdev-5** [comment]
@JackUrb thanks for the answer. Just FYI I was trying to debug the `demo.py` and if I add some code before checking the connection:
```python
    viz = Visdom()

    startup_sec = 1
    while not viz.check_connection() and startup_sec > 0:
        time.sleep(0.1)
        startup_sec -= 0.1

    print("startup_sec:", startup_sec)
    print(viz.win_exists(''), viz.socket_alive, 'or', not viz.use_socket)        
    assert viz.check_connection(), 'No connection could be formed quickly'
```
The assertion is raised and the output for `print(viz.win_exists(''), viz.socket_alive, 'or', not viz.use_socket)` is `False False or False`

**Comment by JackUrb** [comment]
Hi @vfdev-5, I've implemented this in #460. Are you still having the issue caused by #450 when you run from source?

**Comment by vfdev-5** [comment]
@JackUrb let me check this.
Concerning #460 do you plan to merge it ?

**Comment by JackUrb** [comment]
@vfdev-5 soon, want to just get some eyes on it first.

**Comment by vfdev-5** [comment]
@JackUrb tested the master and have no more problems with runing the `demo.py` as reported in the previous comment.

I looked the PR, looks good to me :)

**Comment by JackUrb** [comment]
Implemented in #460
